test/client_test.js

```diff
@@ -1,430 +1,365 @@
+    t.equal(false, val);
+    t.equal(1, n,  'Ensure set is called');
+    t.end();
   });
-}
+});
 
-exports.testDeleteSuccessful = function(beforeExit, assert) {
+test('DeleteSuccessful', function(t) {
   var n = 0;
-  var callbn = 0;
   var dummyServer = new MemJS.Server();
   dummyServer.write = function(requestBuf) {
-    request = MemJS.Utils.parseMessage(requestBuf);
-    assert.equal('hello', request.key);
+    var request = MemJS.Utils.parseMessage(requestBuf);
+    t.equal('hello', request.key.toString());
     n += 1;
     dummyServer.respond({header: {status: 0, opaque: request.header.opaque}});
-  }
+  };
 
   var client = new MemJS.Client([dummyServer]);
   client.delete('hello', function(err, val) {
-    assert.equal(null, err);
-    assert.equal(true, val);
-    callbn += 1;
-  });
-
-  beforeExit(function() {
-    assert.equal(1, n,  'Ensure set is called');
-    assert.equal(1, callbn,  'Ensure callback is called');
+    t.equal(null, err);
+    t.equal(true, val);
+    t.equal(1, n,  'Ensure set is called');
+    t.end();
   });
-}
+});
 
-exports.testDeleteKeyDNE = function(beforeExit, assert) {
+test('DeleteKeyDNE', function(t) {
   var n = 0;
-  var callbn = 0;
   var dummyServer = new MemJS.Server();
   dummyServer.write = function(requestBuf) {
-    request = MemJS.Utils.parseMessage(requestBuf);
-    assert.equal('hello', request.key);
+    var request = MemJS.Utils.parseMessage(requestBuf);
+    t.equal('hello', request.key.toString());
     n += 1;
     dummyServer.respond({header: {status: 1, opaque: request.header.opaque}});
-  }
+  };
 
   var client = new MemJS.Client([dummyServer]);
   client.delete('hello', function(err, val) {
-    assert.equal(null, err);
-    assert.equal(false, val);
-    callbn += 1;
+    t.equal(null, err);
+    t.equal(false, val);
+    t.equal(1, n,  'Ensure set is called');
+    t.end();
   });
+});
 
-  beforeExit(function() {
-    assert.equal(1, n,  'Ensure set is called');
-    assert.equal(1, callbn,  'Ensure callback is called');
-  });
-}
-
-exports.testFlush = function(beforeExit, assert) {
+test('Flush',  function(t) {
   var n = 0;
-  var callbn = 0;
   var dummyServer = new MemJS.Server();
-  dummyServer.host = "example.com";
+  dummyServer.host = 'example.com';
   dummyServer.port = 1234;
   dummyServer.write = function(requestBuf) {
-    request = MemJS.Utils.parseMessage(requestBuf);
-    assert.equal(0x08, request.header.opcode);
+    var request = MemJS.Utils.parseMessage(requestBuf);
+    t.equal(0x08, request.header.opcode);
     n += 1;
     dummyServer.respond({header: {status: 1, opaque: request.header.opaque}});
-  }
+  };
 
   var client = new MemJS.Client([dummyServer, dummyServer]);
   client.flush(function(err, results) {
-    assert.equal(null, err);
-    assert.equal(true, results['example.com:1234']);
+    t.equal(null, err);
+    t.equal(true, results['example.com:1234']);
+    t.equal(2, n, 'Ensure flush is called'); // TODO twice now but once before?
```

Comment:
We do expect it to happen twice -- once for each server. It happens twice in the previous tests as well, but that just wasn't tested. It's good to add this test.